pkg/thanos/statefulset.go

```diff
@@ -259,6 +259,19 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 		})
 	}
 
+	if tr.Spec.GRPCServerTLSConfig != nil {
+		tls := tr.Spec.GRPCServerTLSConfig
+		if tls.CertFile != "" {
```

Comment:
What happens if one of these fields is empty, should we just let it error out on thanos ruler side?

Reply:
Yes, I was thinking we can just pass along the parameters as they are.  I wasn't sure how thanos would handle a partial set of parameters, so I didn't yet add any validation.

Reply:
Ruler checks that

Reply:
Then this looks good to me, as thanos validates we don't want to do any validations.

Reply:
All components use this: https://github.com/thanos-io/thanos/blob/55cb8ca38b3539381dc6a781e637df15c694e50a/pkg/tls/options.go#L17